e2e/test/helpers/RetryOperationHelper.cs

```diff
@@ -51,5 +52,42 @@ public static async Task RetryOperationsAsync(Func<Task> asyncOperation, IRetryP
             }
             while (shouldRetry);
         }
+
+        /// <summary>
+        /// Rety an async operation based on the retry strategy supplied.
+        /// </summary>
+        /// <param name="asyncOperation">The async operation to be retried.</param>
+        /// <param name="retryPolicy">The retry policy to be applied.</param>
+        /// <param name="retryableStatusCodes">The errors to be retried on.</param>
+        /// <param name="logger">The <see cref="MsTestLogger"/> instance to be used.</param>
+        public static async Task RetryOperationsAsync1(Func<Task> asyncOperation, IRetryPolicy retryPolicy, HashSet<IotHubErrorCode> retryableStatusCodes, MsTestLogger logger)
```

Comment:
Can a user's cancellation token be passed along and checked throughout retries?

Reply:
No, neither for the original `RetryOperationsAsync()` call.
https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/e2e/test/helpers/RetryOperationHelper.cs#L25

Reply:
Do we wanna this happen?

Reply:
I would think yes, we'd want retries to stop if the user's cancellation token has been signaled.

Reply:
Maybe it is already being checked by the operation, but we do have code in here to wait an interval, and we'd want that to cancel sooner.

Reply:
Feel free to take it as a task for another PR. Just please add it to a PBI with enough context for someone to be able to solve.

Reply:
added a task [here](https://msazure.visualstudio.com/One/_boards/board/t/IoT-Developers%20and%20Devices-Managed%20SDKs/Backlog%20items/?workitem=15520439)